Sem. 02/task06.cpp

```diff
@@ -0,0 +1,44 @@
+#include <iostream>
+#include <fstream>
+
+using std::cin;
+using std::cout;
+using std::endl;
+using std::ofstream;
+using std::ifstream;
+
+//task06
+void printSourceCode(const char fileName[])
+{
+	ifstream in(fileName);
+
+	if (!in.is_open())
+	{
+		cout << "Error";
+		return;
+	}
+
+	while (!in.eof())
+	{
+		char buffer[1024];
+		in.getline(buffer, 1024);
+		cout << buffer << endl;
+	}
+
+	in.close();
+}
+
+int main6()
+{
+	ifstream in("Task01.cpp");
```

Comment:
You are opening the same stream here and on line 13. Either open the stream only in printSourceCode() or pass the stream as an argument to it.